src/main.rs

```diff
@@ -52,7 +60,23 @@ fn check_file(arg: &str) {
             chname,
             node.kind()
         )
-        .with_source_code(NamedSource::new(arg, src.clone()));
+        .with_source_code(NamedSource::new(&filename, src.clone()));
         println!("{:?}", report);
     }
 }
+
+// Tree-sitter grammars include some configurations to help decide whether the language applies to
+// a given file.
+// Unfortunately, neither the language-detection algorithm nor the configurations are included in
+// the Rust crates. So for now we have a simplified language-detection with hard-coded
+// configurations.
+// See https://tree-sitter.github.io/tree-sitter/syntax-highlighting#language-detection
+fn detect_language(path: &Path) -> Option<tree_sitter::Language> {
```

Comment:
I think we probably want to expose this as a default config, rather than "language detection". We already merged support for glob-specifying how to match files. So I think we want to express this as default values for those fields. Such as `**/*.py` for Python etc.

We can merge this in the meantime until we got the config stuff up and running however.